src/compat-msvc-enter.h

```diff
@@ -0,0 +1,16 @@
+#ifndef TIMESCALEDB_COMPAT_MSVC_ENTER_H
+#define TIMESCALEDB_COMPAT_MSVC_ENTER_H
+
+/*
+* Not all exported symbols in PostgreSQL are marked with PGDLLIMPORT, which causes
+* errors during linking. This hack turns all extern symbols into properly exported
+* symbols so we can use them in our code. Only necessary for files that use these
+* incorrectly unlabeled symbols (e.g., extension.c)
```

Comment:
Could you add to the comment that this is necessary in windows only for data symbols. i.e. variables. Not for functions. Just so we remember what the issue was better.